packages/admin-ui/client/pages/Item/index.js

```diff
@@ -274,83 +275,119 @@ const ItemNotFound = ({ adminPath, errorMessage, list }) => (
   </PageError>
 );
 
+const PreLoadFields = ({ children, list, loadField }) => (
+  <Suspense fallback={<PageLoading />}>
+    <Render>
+      {() => {
+        const initialisationPromises = list.fields
+          .map((item, index) => {
+            try {
+              loadField(item, index, list);
+            } catch (loadingPromiseOrError) {
+              // An actual error was thrown, so we want to bubble that up
+              if (!isPromise(loadingPromiseOrError)) {
+                throw loadingPromiseOrError;
+              }
+              // Return a Suspense promise
+              return loadingPromiseOrError;
+            }
+          })
+          .filter(Boolean);
+
+        if (initialisationPromises.length) {
+          // Trigger the suspense boundary and wait for all fields in
+          // parallel.
+          throw Promise.all(initialisationPromises);
+        }
+
+        return children();
+      }}
+    </Render>
+  </Suspense>
+);
```

Comment:
In a future PR, I will move this component somewhere re-usable as the logic is identical for `Cell`/`Filter`.